[CI] RestEsqlIT testInternalRange {ASYNC} failing

**Build Scans:**
- [elasticsearch-periodic #6046 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/af3wdhkdrwrha)
- [elasticsearch-intake #16489 / part3](https://gradle-enterprise.elastic.co/s/lrfmmjvlold4y)
- [elasticsearch-periodic-platform-support #6035 / ubuntu-1804_platform-support-unix](https://gradle-enterprise.elastic.co/s/zecakfyfn2gr4)
- [elasticsearch-periodic-platform-support #6035 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/zndnea2h4ba6e)
- [elasticsearch-periodic-platform-support #6035 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/cpjlyag32j35o)
- [elasticsearch-periodic #6046 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/ffrbgb7r2vwra)
- [elasticsearch-intake #16487 / part3](https://gradle-enterprise.elastic.co/s/pssaskf7qjhjw)
- [elasticsearch-periodic-platform-support #6035 / almalinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/ymihwipwo52vo)
- [elasticsearch-intake #16481 / part3](https://gradle-enterprise.elastic.co/s/sv2ex3iqtipxq)
- [elasticsearch-intake #16480 / part3](https://gradle-enterprise.elastic.co/s/bkpridgupv3zy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.RestEsqlIT.testInternalRange {ASYNC}" -Dtests.seed=C12CE00B1A4EEF0D -Dtests.locale=ff-Adlm-BF -Dtests.timezone=Etc/GMT+9 -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.RestEsqlIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testInternalRange%20%7BASYNC%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [GET], host [http://[::1]:43227], URI [/_query/async/FkNjNEhBR2JHU09tUlNKWFY1MDN1ZWcbOFNLdmx4US1TVmF4amxUend2Q0pkUToyMzAy?wait_for_completion_timeout=60s&pretty=true&error_trace=true], status line [HTTP/1.1 502 Bad Gateway]
¿eerror¿jroot_cause¿dtypexnode_not_connected_exceptionfreasonx;[test-cluster-1][127.0.0.1:45271] connection already closedkstack_tracexsorg.elasticsearch.transport.NodeNotConnectedException: [test-cluster-1][127.0.0.1:45271] connection already closed
ÿÿdtypexnode_not_connected_exceptionfreasonx;[test-cluster-1][127.0.0.1:45271] connection already closedkstack_tracexsorg.elasticsearch.transport.NodeNotConnectedException: [test-cluster-1][127.0.0.1:45271] connection already closed
ÿfstatusöÿ
```

**Issue Reasons:**
- [8.x] 4 consecutive failures in test testInternalRange {ASYNC}
- [8.x] 2 consecutive failures in step part3
- [8.x] 2 consecutive failures in pipeline elasticsearch-intake
- [8.x] 19 failures in test testInternalRange {ASYNC} (10.7% fail rate in 178 executions)
- [8.x] 9 failures in step part3 (39.1% fail rate in 23 executions)
- [8.x] 2 failures in step part-3 (4.1% fail rate in 49 executions)
- [8.x] 9 failures in pipeline elasticsearch-intake (39.1% fail rate in 23 executions)
- [8.x] 2 failures in pipeline elasticsearch-pull-request (4.1% fail rate in 49 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Seems to be another one of those. /cc @dnhatn 

```
[2025-01-28T11:21:52,682][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-0] fatal error in thread [elasticsearch[test-cluster-0][generic][T#3]], exiting java.lang.AssertionError: java.lang.AssertionError: seqNo [100] was processed twice in generation [2], with different data. prvOp [Index{id='100', seqNo=100, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}], newOp [Index{id='100', seqNo=100, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}]	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.assertNoSeqNumberConflict(TranslogWriter.java:304)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.add(TranslogWriter.java:264)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.Translog.add(Translog.java:629)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.engine.InternalEngine.index(InternalEngine.java:1234)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.index(IndexShard.java:1085)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.applyIndexOperation(IndexShard.java:1011)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.applyTranslogOperation(IndexShard.java:1991)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.applyTranslogOperation(IndexShard.java:1978)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.RecoveryTarget.lambda$indexTranslogOperations$4(RecoveryTarget.java:454)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:356)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.RecoveryTarget.indexTranslogOperations(RecoveryTarget.java:429)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.PeerRecoveryTargetService$TranslogOperationsRequestHandler.performTranslogOps(PeerRecoveryTargetService.java:658)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.PeerRecoveryTargetService$TranslogOperationsRequestHandler.handleRequest(PeerRecoveryTargetService.java:605)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.PeerRecoveryTargetService$TranslogOperationsRequestHandler.handleRequest(PeerRecoveryTargetService.java:597)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.PeerRecoveryTargetService$RecoveryRequestHandler.messageReceived(PeerRecoveryTargetService.java:691)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.indices.recovery.PeerRecoveryTargetService$RecoveryRequestHandler.messageReceived(PeerRecoveryTargetService.java:678)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:76)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleRequest(InboundHandler.java:289)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:302)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1044)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)	
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)	
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)	
	at java.base/java.lang.Thread.run(Thread.java:1575)	
Caused by: java.lang.AssertionError: seqNo [100] was processed twice in generation [2], with different data. prvOp [Index{id='100', seqNo=100, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}], newOp [Index{id='100', seqNo=100, primaryTerm=1, version=1, autoGeneratedIdTimestamp=-1}]	
	... 24 more	
Caused by: java.lang.RuntimeException: stack capture previous op	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.assertNoSeqNumberConflict(TranslogWriter.java:310)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.TranslogWriter.add(TranslogWriter.java:264)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.translog.Translog.add(Translog.java:629)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.engine.InternalEngine.index(InternalEngine.java:1234)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.index(IndexShard.java:1085)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.applyIndexOperation(IndexShard.java:1011)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.index.shard.IndexShard.applyIndexOperationOnReplica(IndexShard.java:952)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.bulk.TransportShardBulkAction.performOpOnReplica(TransportShardBulkAction.java:697)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.bulk.TransportShardBulkAction.performOnReplica(TransportShardBulkAction.java:674)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.bulk.TransportShardBulkAction.lambda$dispatchedShardOperationOnReplica$5(TransportShardBulkAction.java:630)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.ActionListener.completeWith(ActionListener.java:356)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnReplica(TransportShardBulkAction.java:628)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnReplica(TransportShardBulkAction.java:82)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.action.support.replication.TransportWriteAction$2.doRun(TransportWriteAction.java:248)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)	
	at org.elasticsearch.server@8.18.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:34)	
	... 5 more
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -255,11 +255,6 @@ tests:
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test070BindMountCustomPathConfAndJvmOptions
   issue: https://github.com/elastic/elasticsearch/issues/120910
-- class: org.elasticsearch.xpack.esql.qa.multi_node.RestEsqlIT
-  issue: https://github.com/elastic/elasticsearch/issues/120948
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level}
-  issue: https://github.com/elastic/elasticsearch/issues/120950
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test071BindMountCustomPathWithDifferentUID
   issue: https://github.com/elastic/elasticsearch/issues/120918
@@ -283,9 +278,6 @@ tests:
 - class: org.elasticsearch.xpack.security.profile.ProfileIntegTests
   method: testSuggestProfilesWithName
   issue: https://github.com/elastic/elasticsearch/issues/121022
-- class: org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT
-  method: testBulkOperations {p0=true}
-  issue: https://github.com/elastic/elasticsearch/issues/120969
 
 # Examples:
 #

--- a/server/src/main/java/org/elasticsearch/index/engine/TranslogDirectoryReader.java
+++ b/server/src/main/java/org/elasticsearch/index/engine/TranslogDirectoryReader.java
@@ -97,7 +97,7 @@ final class TranslogDirectoryReader extends DirectoryReader {
             // When using synthetic source, the translog operation must always be reindexed into an in-memory Lucene to ensure consistent
             // output for realtime-get operations. However, this can degrade the performance of realtime-get and update operations.
             // If slight inconsistencies in realtime-get operations are acceptable, the translog operation can be reindexed lazily.
-            if (mappingLookup.isSourceSynthetic()) {
+            if (mappingLookup.isSourceSynthetic() || mappingLookup.inferenceFields().isEmpty() == false) {
                 onSegmentCreated.run();
                 leafReader = createInMemoryReader(shardId, engineConfig, directory, documentParser, mappingLookup, false, operation);
             } else {

--- a/server/src/main/java/org/elasticsearch/index/engine/TranslogOperationAsserter.java
+++ b/server/src/main/java/org/elasticsearch/index/engine/TranslogOperationAsserter.java
@@ -38,7 +38,8 @@ public abstract class TranslogOperationAsserter {
                 if (super.assertSameIndexOperation(o1, o2)) {
                     return true;
                 }
-                if (engineConfig.getIndexSettings().isRecoverySourceSyntheticEnabled()) {
+                if (engineConfig.getIndexSettings().isRecoverySourceSyntheticEnabled()
+                    || engineConfig.getMapperService().mappingLookup().inferenceFields().isEmpty() == false) {
                     return super.assertSameIndexOperation(synthesizeSource(engineConfig, o1), o2)
                         || super.assertSameIndexOperation(o1, synthesizeSource(engineConfig, o2));
                 }
@@ -60,23 +61,39 @@ public abstract class TranslogOperationAsserter {
                 TrivialQueryCachingPolicy.NEVER,
                 () -> {}
             );
-            try (
-                LuceneSyntheticSourceChangesSnapshot snapshot = new LuceneSyntheticSourceChangesSnapshot(
-                    engineConfig.getMapperService(),
-                    searcher,
-                    LuceneSyntheticSourceChangesSnapshot.DEFAULT_BATCH_SIZE,
-                    Integer.MAX_VALUE,
-                    op.seqNo(),
-                    op.seqNo(),
-                    true,
-                    false,
-                    engineConfig.getIndexSettings().getIndexVersionCreated()
-                )
-            ) {
+            try (var snapshot = newSnapshot(engineConfig, op, searcher);) {
                 final Translog.Operation normalized = snapshot.next();
                 assert normalized != null : "expected one operation; got zero";
                 return (Translog.Index) normalized;
             }
+        }
+    }
+
+    static Translog.Snapshot newSnapshot(EngineConfig engineConfig, Translog.Index op, Engine.Searcher searcher) throws IOException {
+        if (engineConfig.getIndexSettings().isRecoverySourceSyntheticEnabled()) {
+            return new LuceneSyntheticSourceChangesSnapshot(
+                engineConfig.getMapperService(),
+                searcher,
+                LuceneSyntheticSourceChangesSnapshot.DEFAULT_BATCH_SIZE,
+                Integer.MAX_VALUE,
+                op.seqNo(),
+                op.seqNo(),
+                true,
+                false,
+                engineConfig.getIndexSettings().getIndexVersionCreated()
+            );
+        } else {
+            return new LuceneChangesSnapshot(
+                engineConfig.getMapperService(),
+                searcher,
+                LuceneSyntheticSourceChangesSnapshot.DEFAULT_BATCH_SIZE,
+                op.seqNo(),
+                op.seqNo(),
+                true,
+                false,
+                false,
+                engineConfig.getIndexSettings().getIndexVersionCreated()
+            );
         }
     }
 

